conans/client/build/cmake_toolchain_build_helper.py

```diff
@@ -75,7 +75,8 @@ def configure(self, source_folder=None):
             build_folder = os.path.join(self._conanfile.build_folder, self._build_folder)
 
         mkdir(build_folder)
-        arg_list = '-DCMAKE_TOOLCHAIN_FILE="%s" "%s"' % (CMakeToolchainBase.filename, source)
+        arg_list = '-DCMAKE_TOOLCHAIN_FILE="{}" -DCMAKE_INSTALL_PREFIX="{}" "{}"'.format(
+            CMakeToolchainBase.filename, self._conanfile.package_folder, source)
```

Comment:
Probably add the ``.package_folder.replace("\\", "/")`` that was there? It might be necessary in Windows?

Reply:
In the old build-helper we are not running the replacement (https://github.com/conan-io/conan/blob/e6385bbf639898400a10841cee37910f4fe10a38/conans/client/build/cmake_flags.py#L352)... but now I see we are setting many other variables. Should I set all of them? maybe use `GNUInstallDirs`?

Reply:
It seems ``GNUInstallDirs`` define the system ones, not the local ones, which sounds unwanted.
Passing all of them in the command line reads very ugly again. What is currently defined:

```python
definitions["CMAKE_INSTALL_BINDIR"] = DEFAULT_BIN
definitions["CMAKE_INSTALL_SBINDIR"] = DEFAULT_BIN
...
definitions["CMAKE_INSTALL_OLDINCLUDEDIR"] = DEFAULT_INCLUDE
definitions["CMAKE_INSTALL_DATAROOTDIR"] = DEFAULT_SHARE
```

Is also a bit wrong, as it is hardcoding some values, irrespective of what is defined in the ``package_info()`` and ``package()`` method. I intentionally left this out until we had some time to think about it. Shouldn't those values be defined by the library ``CMakeLists.txt``?